docker/notes.txt

```diff
@@ -0,0 +1,11 @@
+# the commands below should be run from the repo base (.. from this file)
+
+# build the docker image. first time will take longer due to getting dependencies.
+docker build -t test -f docker/Dockerfile .
+
+# run the image, using bash, rm when done
+docker run --publish 2015:2015 -it --rm test:latest /bin/bash
```

Comment:
What does this do?  Is it necessary to run bash first to run a go server?  My lack of docker knowledge is showing... 🤦

Reply:
Both of these are just so you can interact with the container while it's running. Neither is necessary to just run the container. This could instead be:
docker run --name PaoServer -d -p 2015:2015 arbrown/pao:latest

Which will run the container named "PaoServer" mapping container port 2015 to system port 2015 from latest version of the image named "arbrown/pao".

Reply:
I also found that running these interactively didn't work, but using detached they launched just fine.

Reply:
The `docker run ... /bin/bash` is just to be able to interact with the image from a shell prompt.  I find it useful when creating and modifying images. It does not modify the image in any way and wouldn’t be used in normal workflow.

Reply:
The last `docker run ...` command actually runs the server and you can then hit `local host:2015` to do it.  I usually run it interactively (in its own terminal) and just use `^C` when done. Detached `docker run  —detached ... ` also works, it’s just more typing to kill it.